grails-app/services/org/broadinstitute/orsp/QueryService.groovy

```diff
@@ -725,14 +727,65 @@ class QueryService implements Status {
 
         def rows = getSqlConnection().rows(query, params)
         def ids = rows.collect{it.get("id")}
-        Collection result = Collections.emptyList()
+        Set result = new HashSet<Issue>()
 
         if (ids.size() > 0) {
-            result = Issue.findAllByIdInList(ids)
+            result = findIssuesSearchItemsDTO(ids)
         }
         result
     }
 
+    /**
+     * Find issues and create DTOs to narrow the amount of data used in the search process
+     * @param issueIds
+     * @return
+     */
+    Set<IssueSearchItemDTO> findIssuesSearchItemsDTO(ArrayList<Integer> issueIds) {
+        final String query = "SELECT i.id id, " +
+                "i.project_key projectKey, " +
+                "i.type type, " +
+                "i.status status, " +
+                "i.summary summary, " +
+                "i.reporter reporter, " +
+                "i.update_date updated, " +
+                "i.expiration_date expirationDate, " +
+                "iep.* " +
+                "FROM issue i LEFT JOIN issue_extra_property iep " +
+                "ON (iep.project_key = i.project_key AND iep.name in ('pm','pi','collaborator')) " +
+                "WHERE i.id IN (" + issueIds.join(",") + ")"
+
+        Set<IssueSearchItemDTO> resultDTO = new HashSet<IssueSearchItemDTO>()
+        IssueSearchItemDTO issueSearchItemDTO
+        String currentProjectKey = ""
+
+        getSqlConnection().rows(query).each {
+            if (it.get("projectKey") == currentProjectKey) {
```

Comment:
All of this logic should be in a constructor for `IssueSearchItemDTO`. That class should accept a `GroovyRowResult` as an argument (that's what `it` is in this line) and perform all of the logic there. Having this in the query service breaks encapsulation and makes it impossible to re-use `IssueSearchItemDTO` in any other context.

Reply:
Agree.